aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,623 @@
+        assert!(proposal.resolution == PROPOSAL_UNRESOLVED, error::invalid_argument(EPROPOSAL_RESOLVED));
+
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+            return
+        };
+        let passed = if (voting_stat.total_yes > voting_stat.total_no) {true} else {false};
+        if (passed) {
+            let accts = &borrow_global<RegisteredDAOs>(@dao_platform).res_acct_to_src_acct;
+            assert!(table::contains(accts, nft_dao), error::not_found(ERESOURCE_ACCT_NOT_EXIST));
+            let src_acct = *table::borrow(accts, nft_dao);
+
+            let function_name = proposal.function_name;
+            if (function_name == string::utf8(b"transfer_fund")) {
+                // This is very dangerous.
+                // We should exploring have server side dynamic compiling and deploying DAO contracts in each DAO's own accounts.
+                let res_signer = get_dao_signer(src_acct);
+                let dst_addr = property_map::read_address(&proposal.function_args, &string::utf8(b"dst"));
+                let amount = property_map::read_u64(&proposal.function_args, &string::utf8(b"amount"));
+                transfer_fund(&res_signer, dst_addr, amount);
+            } else {
+               assert!(function_name == string::utf8(b"no_op"), error::invalid_argument(ENOT_SUPPROTED_FUNCTION));
+            };
+            proposal.resolution = PROPOSAL_RESOLVED_PASSED;
+        } else {
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        };
+    }
+
+    /// DAO creator can delegate its DAO to another admin account
+    public entry fun delegate_dao(admin: &signer, new_admin: address)acquires RegisteredDAOs {
+        let registered_accounts = borrow_global_mut<RegisteredDAOs>(@dao_platform);
+        let old_addr = signer::address_of(admin);
+        assert!(table::contains(&registered_accounts.accounts_signer_caps, old_addr), error::not_found(ENOT_FOUND_ADMIN_ACCOUNT));
+        assert!(!table::contains(&registered_accounts.accounts_signer_caps, old_addr), error::not_found(EDELEGATED_ADDRESS_ALREADY_EXISTS));
+
+        let signer_cap = table::remove(&mut registered_accounts.accounts_signer_caps, old_addr);
+        let res_addr = get_signer_capability_address(&signer_cap);
+        table::add(&mut registered_accounts.accounts_signer_caps, new_admin, signer_cap);
+
+        let src_addr = table::borrow_mut(&mut registered_accounts.res_acct_to_src_acct, res_addr);
+        *src_addr = new_admin;
+
+    }
+
+    /// DAO creator can quit the platform and claim back his resource account signer capability
```

Comment:
Isn't this action equivalent to overtaking the DAO since proposal actions such as transfer would no longer work? 🤔 Are there any precautions or cleanups we need to do here if that's the case?

Reply:
this is for quitting the platform and get back the signer capability that is stored within the platform